### ISSUE TITLE
Training 3D - Root relative location maps

### ISSUE THREAD

**Issue by designpirate** [hwangdonghyun/VNect-TensorFlow-Keras#4]
Hi there. I acquired a dataset that has accurate joint positions in a different coordinate space that I want to use to train this model. I transformed the joints to be root-relative (to the pelvis) as I understood from the vnect paper. However, I am struggling to understand how one should produce the location maps. I understand x and y fairly well, but I am a bit confused what I should do with z outside of the region that would be covered by the heatmap. The loss function should send everything to 0 outside of that region, but am I missing something? Do you have any guidance?

Furthermore, I successfully trained the 2D heatmap already (though I did it with individual points for the joints, not a gaussian distribution) - when you train for 3D, do you include the 2D heatmap as a loss term with mean squared error again - and should the 2d heatmap be created using a gaussian? The paper alludes to this but it is not explicitly stated.

Thanks in advance.

**Comment by hwangdonghyun** [comment]
First, sorry for the late reply to this issue.
If the network can't detect 2D joints then we can't catch the value of 3D joints positions.
When the training, if 2D GT is outside of image 3D GT will be calculated as zero based on Hadamard production.
Maybe loss can not be zero if your network detects a false-positive one. 

I give 2D and 3D losses simultaneously for the 3D dataset. The gaussian heatmap will be fine.
